src/Storages/PostgreSQL/PostgreSQLReplicationHandler.cpp

```diff
@@ -279,7 +281,9 @@ ASTPtr PostgreSQLReplicationHandler::getCreateNestedTableQuery(StorageMaterializ
     auto table_structure = std::make_unique<PostgreSQLTableStructure>(fetchPostgreSQLTableStructure(tx, table_name, postgres_schema, true, true, true));
     if (!table_structure)
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Failed to get PostgreSQL table structure");
-    return storage->getCreateNestedTableQuery(std::move(table_structure));
+
+    auto table_override = tryGetTableOverride(current_database_name, table_name);
+    return storage->getCreateNestedTableQuery(std::move(table_structure), table_override ? table_override->as<ASTTableOverride>() : nullptr);
```

Comment:
The `?:` should not be needed here since `tryGetTableOverride()` returns nullptr when not found.

Reply:
Nevermind, I ignored the cast :P